pssh/ssh_client.py

```diff
@@ -100,6 +100,14 @@ def __init__(self, host,
                             else host)
         _user = host_config['user'] if 'user' in host_config else None
         user = user if user else _user
+
+        _port = int(host_config['port']) if 'port' in host_config else 22
+        port = port if port else _port
+
+        # FIXME: we take the first key
+        _pkey = paramiko.RSAKey.from_private_key_file(host_config['identityfile'][0]) if 'identityfile' in host_config else None
```

Comment:
What if key is not RSA? This likely needs to try both key types and catch exceptions for the wrong key type.